log4j-plugins/src/main/java/org/apache/logging/log4j/plugins/inject/PluginAttributeParser.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+
+package org.apache.logging.log4j.plugins.inject;
+
+import org.apache.logging.log4j.plugins.Inject;
+import org.apache.logging.log4j.plugins.Node;
+import org.apache.logging.log4j.plugins.PluginAttribute;
+import org.apache.logging.log4j.plugins.convert.TypeConverters;
+import org.apache.logging.log4j.plugins.name.AnnotatedElementAliasesProvider;
+import org.apache.logging.log4j.plugins.name.AnnotatedElementNameProvider;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Parameter;
+import java.lang.reflect.Type;
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+
+public class PluginAttributeParser implements NodeParser<PluginAttribute> {
+    private static final Map<Type, Function<PluginAttribute, ?>> DEFAULT_VALUE_EXTRACTORS = Map.ofEntries(
+            Map.entry(int.class, PluginAttribute::defaultInt),
+            Map.entry(Integer.class, PluginAttribute::defaultInt),
+            Map.entry(long.class, PluginAttribute::defaultLong),
+            Map.entry(Long.class, PluginAttribute::defaultLong),
+            Map.entry(boolean.class, PluginAttribute::defaultBoolean),
+            Map.entry(Boolean.class, PluginAttribute::defaultBoolean),
+            Map.entry(float.class, PluginAttribute::defaultFloat),
+            Map.entry(Float.class, PluginAttribute::defaultFloat),
+            Map.entry(double.class, PluginAttribute::defaultDouble),
+            Map.entry(Double.class, PluginAttribute::defaultDouble),
+            Map.entry(byte.class, PluginAttribute::defaultByte),
+            Map.entry(Byte.class, PluginAttribute::defaultByte),
+            Map.entry(char.class, PluginAttribute::defaultChar),
+            Map.entry(Character.class, PluginAttribute::defaultChar),
+            Map.entry(short.class, PluginAttribute::defaultShort),
+            Map.entry(Short.class, PluginAttribute::defaultShort),
+            Map.entry(Class.class, PluginAttribute::defaultClass)
+    );
+
+    private final Function<String, String> stringSubstitutionStrategy;
+
+    @Inject
```

Comment:
Note how the parsing strategies for nodes can themselves be dependency-injected. In this case, we're using a `StrSubstitutor` function.